src/main/java/seedu/address/logic/commands/editcommand/EditParticipantCommand.java

```diff
@@ -0,0 +1,132 @@
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Phone;
+
+public class EditParticipantCommand extends EditCommand {
+
+    /* Possible Fields */
+    public static final String MESSAGE_EDIT_PARTICIPANT_SUCCESS = "Edited Participant: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided."; // to be used in Parser
+    public static final String MESSAGE_DUPLICATE_PARTICIPANT = "This person already exists in the address book.";
+    public static final String MESSAGE_INVALID_PARTICIPANT_DISPLAYED_INDEX = "The participant index provided is invalid";
+
+    private EditParticipantDescriptor editParticipantDescriptor;
+
+    public EditParticipantCommand(Id id, EditParticipantDescriptor editParticipantDescriptor) {
+        super(id);
+        requireNonNull(editParticipantDescriptor);
+        this.editParticipantDescriptor = editParticipantDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Participant participantToEdit;
+        try {
+            participantToEdit = model.getParticipant(this.id);
+        } catch (AlfredException e) {
+            throw new CommandException(MESSAGE_INVALID_PARTICIPANT_DISPLAYED_INDEX);
+        }
+        Participant editedParticipant = this.createEditedParticipant(participantToEdit,
+                                                                     this.editParticipantDescriptor);
+
+        // Model should check if there exists duplicates in list
+        /*
+         * i.e
+         * if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+         *     throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+         * }
+         */
+        model.updateParticipant(this.id, editedParticipant);
+//        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_PARTICIPANT_SUCCESS, editedParticipant.toString()));
+    }
+
+    private Participant createEditedParticipant(Participant participantToEdit,
+                                                EditParticipantDescriptor editParticipantDescriptor) {
+        assert participantToEdit != null;
+
+        Name updatedName = editParticipantDescriptor.getName().orElse(participantToEdit.getName());
+        Id updatedId = editParticipantDescriptor.getId().orElse(participantToEdit.getId());
+        Email updatedEmail = editParticipantDescriptor.getEmail().orElse(participantToEdit.getEmail());
+        Phone updatedPhone = editParticipantDescriptor.getPhone().orElse(participantToEdit.getPhone());
+
+        // Reorder parameters as necessary
+        return new Participant(updatedName, updatedEmail, updatedPhone, updatedId);
```

Comment:
Will reorder the params on my end as Name, Id, email, phone based on your prev comment. Sorry 😓